src/Corax/Queries/TermMatch.cs

```diff
@@ -120,17 +122,24 @@ static int FillFunc(ref TermMatch term, Span<long> matches)
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
             static int AndWithFunc(ref TermMatch term, Span<long> buffer, int matches)
             {
-                // TODO: If matches is too big, we should use quicksort
+                uint bot = 0;
+                uint top = (uint)matches;
+
                 long current = term._current;
-                for (int i = 0; i < matches; i++)
+                while (top > 1)
                 {
-                    if (buffer[i] == current)
-                    {
-                        buffer[0] = current;
-                        return 1;
-                    }
+                    uint mid = top / 2;
+
+                    if (current >= Unsafe.Add(ref MemoryMarshal.GetReference(buffer), bot + mid))
+                        bot += mid;
+                    top -= mid;
                 }
-                return 0;
+
+                if (current != Unsafe.Add(ref MemoryMarshal.GetReference(buffer), bot))
+                    return 0;
+
```

Comment:
Is there a guarantee that ids in the buffer are sorted?

Reply:
Yes.